### DISCUSSION TITLE
Do/Undo commands to enable/disable display?

### DISCUSSION THREAD

**Discussion by mdrichardson** [LizardByte/community#288] [Sunshine]
My PC is on an Ultrawide monitor and I use Moonlight on a TV, so I use [IddSampleDriver](https://github.com/ge9/IddSampleDriver) to create a virtual monitor.

However, I rarely use Moonlight (once every couple weeks), so it would be nice to have that virtual monitor disabled and then only enabled for Moonlight streams.

I've added CLI commands (that work just fine) as Do/Undo commands at both the app and global config levels. The problem is that when trying to connect via Moonlight, it gives the error about not being able to initialize the display, like it's trying to do that before the Do command runs.

Is there a better/different way to do this?

**Comment by ReenigneArcher** [comment]
Prep commands run before the stream starts... So my guess is that your script is completed before the screen is actually initialized. Maybe add a sleep to your script for a few seconds.

**Reply by mdrichardson** [reply]
Ah, I'll try that. Currently, it's just a command: `MultiMonitorTool.exe /enable 4`.

Are the prep Do commands executed synchronously, top to bottom? Can I just add an additional command, `powershell -Command Start-Sleep 5`?

**Reply by mdrichardson** [reply]
Also, I _think_, based on the logs, that if Sunshine doesn't see the display on startup, it's going to have an issue. Is there a way to get Sunshine to re-check once it actually needs it, if it doesn't already do so?

**Reply by ReenigneArcher** [reply]
Yea, an additional command should work.

You are probably right about if it's not present at startup. That behavior may be improved in the nightly release though (not 100% sure).

**Reply by mdrichardson** [reply]
> Prep commands run before the stream starts

Bummer. Looks like this is not actually the case. I made a simple `do` command of `cmd /C "echo do"`:

1. With the monitor disabled, launching an app via Moonlight, I don't see the command prompt pop up on the host
2. With the monitor enabled, launching an app via Moonlight, the command prompt pops up

The logs seem to agree...it shows the display information before executing the `do` command.

**Reply by dna113p** [reply]
Did you ever solve this issue?

**Reply by mdrichardson** [reply]
@dna113p It became part of [this discussion](https://github.com/LizardByte/Sunshine/discussions/1553). I'm currently using @Nonary's script and the primary monitor switching, though I'd prefer to just be able to run Do commands prior to trying to connect to the display adapter.

[a comment by an automated account is not shown]
